core/src/test/kotlin/pyobject/TestPyString.kt

```diff
@@ -17,24 +17,304 @@
 
 package green.sailor.kython.test.pyobject
 
+import green.sailor.kython.interpreter.pyobject.PyBool
 import green.sailor.kython.interpreter.pyobject.PyString
-import green.sailor.kython.test.assertUnwrappedTrue
-import green.sailor.kython.test.testWithObject
+import green.sailor.kython.interpreter.pyobject.PyTuple
+import green.sailor.kython.test.testPrimitive
+import green.sailor.kython.util.center
+import java.util.*
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
 import org.junit.jupiter.params.ParameterizedTest
 import org.junit.jupiter.params.provider.ValueSource
 
 class TestPyString {
     @ParameterizedTest(name = "For value {0}")
     @ValueSource(strings = ["test", "randomString", "123", "", "1Hello", "Hey!"])
     fun `Test str lower`(value: String) {
-        val result = testWithObject<PyString>("""result = str("$value").lower()""")
-        assertUnwrappedTrue(result) { it == value.toLowerCase() }
+        testPrimitive<PyString>("""result = str("$value").lower()""") {
+            resultsIn(value.toLowerCase())
+        }
     }
 
     @ParameterizedTest(name = "For value {0}")
     @ValueSource(strings = ["test", "randomString", "123", "", "1Hello", "Hey!"])
     fun `Test str upper`(value: String) {
-        val result = testWithObject<PyString>("""result = str("$value").upper()""")
-        assertUnwrappedTrue(result) { it == value.toUpperCase() }
+        testPrimitive<PyString>("""result = str("$value").upper()""") {
+            resultsIn(value.toUpperCase())
+        }
+    }
+
+    @ParameterizedTest(name = "For value {0}")
+    @ValueSource(strings = ["test", "randomString", "123", "", "1Hello", "Hey!"])
+    fun `Test str capitalize`(value: String) {
+        testPrimitive<PyString>("""result = str("$value").capitalize()""") {
+            resultsIn(value.capitalize())
+        }
+    }
+
+    @ParameterizedTest(name = "For value {0}")
+    @ValueSource(strings = ["test", "randomString", "123", "", "Grüße"])
+    fun `Test str casefold`(value: String) {
+        testPrimitive<PyString>("""result = str("$value").casefold()""") {
+            resultsIn(value.toUpperCase(Locale.US).toLowerCase(Locale.US))
+        }
+    }
+
+    @Test
+    fun `Test str center width and fillchar`() {
+        testPrimitive<PyString>("""result = str("moon").center(10, ":")""") {
+            resultsIn("moon".center(10L, ':'))
```

Comment:
This is actually a circular test (since pyStringCenter will just call .center). Test it against a fixed string.